staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -216,7 +263,6 @@ func shutdownInCloudProvider(ctx context.Context, cloud cloudprovider.Interface,
 }
 
 // ensureNodeExistsByProviderID checks if the instance exists by the provider id,
-// If provider id in spec is empty it calls instanceId with node name to get provider id
 func ensureNodeExistsByProviderID(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node) (bool, error) {
 	if instanceV2, ok := cloud.InstancesV2(); ok {
```

Comment:
this is called twice now, here and inside `getProviderID`

Reply:
To save on this `InstancesV2` call I could modify the signature of all 3 functions to be
```
func chocobomb(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node, instancesV2 bool)
```
so that in the whole `MonitorNodes` we would call it only once. That would be even better than what we have today because today we call it twice.

Similarly with `cloud.Instances()`, it could be passed as part of signature e.g.

```
func chocobomb(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node, instancesV1 bool, instancesV2 bool)
```

and again we optimized from 2 calls today to 1 call tomorrow. How do you feel about such an approach?

Reply:
Calling `cloud.InstancesV2()` twice is not a big deal, it usually just returns a field to a struct